src/main.cpp

```diff
@@ -2631,6 +2631,26 @@ static bool ActivateBestChainStep(CValidationState &state, CBlockIndex *pindexMo
     const CBlockIndex *pindexOldTip = chainActive.Tip();
     const CBlockIndex *pindexFork = chainActive.FindFork(pindexMostWork);
 
+    auto reorgLength = pindexOldTip->nHeight - pindexFork->nHeight;
+    if (reorgLength > MAX_REORG_LENGTH) {
+        auto msg = strprintf(_(
+            "A block chain reorganization has been detected that would roll back %d blocks! "
+            "This is larger than the maximum of %d blocks, and so the node is shutting down for your safety."
+            ), reorgLength, MAX_REORG_LENGTH) + "\n\n" +
+            _("Fork details") + ":\n" +
+            "- " + strprintf(_("Current tip: %s, height %d, work %s"),
```

Comment:
"work" is a long string of mostly zeros where any useful information is at the end. Can this be improved (e.g. by stripping leading zeros)? There's no real reason for work to be expressed in hex, right?

Reply:
I went with hex because that's how it is rendered everywhere else, and I chose to render it here because the reorg is being caused by a difference in work (c/f the rewind message which doesn't include work). I'm not sure it matters TBH, as any user seeing this will just copy-paste the message into an issue.

Reply:
Hmm. Can we reduce the line lengths so that the message is more likely to make it intact through such a copy-paste?